build.psm1

```diff
@@ -0,0 +1,291 @@
+
+        $destinationDir = "$projectRoot\out\PSScriptAnalyzer"
+        # this is normalizing case as well as selecting the proper location
+        if ( $Framework -eq "core" ) {
+            $destinationDirBinaries = "$destinationDir\coreclr"
+        }
+        elseif ($PSVersion -eq '3') {
+            $destinationDirBinaries = "$destinationDir\PSv3"
+        }
+        elseif ($PSVersion -eq '4') {
+            $destinationDirBinaries = "$destinationDir\PSv4"
+        }
+        else {
+            $destinationDirBinaries = $destinationDir
+        }
+
+        # build the analyzer
+        #Write-Progress "Building for framework $Framework, configuration $Configuration"
+        # The Rules project has a dependency on the Engine therefore just building the Rules project is enough
+        $config = "PSV${PSVersion}${Configuration}"
+        try {
+            Push-Location $projectRoot/Rules
+            Write-Progress "Building ScriptAnalyzer '$framework' version '${PSVersion}' configuration '${Configuration}'"
+            $buildOutput = dotnet build Rules.csproj --framework $frameworkName --configuration "${config}"
+            if ( $LASTEXITCODE -ne 0 ) { throw "$buildOutput" }
+        }
+        catch {
+            Write-Error "Failure to build $framework ${config}"
+            return
+        }
+        finally {
+            Pop-Location
+        }
+
+        #Write-Progress "Copying files to $destinationDir"
+        Publish-File $itemsToCopyCommon $destinationDir
+
+        $itemsToCopyBinaries = @(
+            "$projectRoot\Engine\bin\${config}\${frameworkName}\Microsoft.Windows.PowerShell.ScriptAnalyzer.dll",
+            "$projectRoot\Rules\bin\${config}\${frameworkName}\Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules.dll"
+            )
+        Publish-File $itemsToCopyBinaries $destinationDirBinaries
+
+        Publish-File $settingsFiles (Join-Path -Path $destinationDir -ChildPath Settings)
+
+        # copy newtonsoft dll if net451 framework
+        if ($Framework -eq "full") {
+            Copy-Item -path "$projectRoot\Rules\bin\${config}\${frameworkName}\Newtonsoft.Json.dll" -Destination $destinationDirBinaries
+        }
+
+        Pop-Location
+    }
+}
+
+# TEST HELPERS
+# Run our tests
+function Test-ScriptAnalyzer
+{
+    [CmdletBinding()]
+    param ( [Parameter()][switch]$InProcess )
+
+    END {
+        $testModulePath = Join-Path "${projectRoot}" -ChildPath out
+        $testResultsFile = Join-Path ${projectRoot} -childPath TestResults.xml
+        $testScripts = "${projectRoot}\Tests\Engine,${projectRoot}\Tests\Rules,${projectRoot}\Tests\Documentation"
+        try {
+            $savedModulePath = $env:PSModulePath
+            $env:PSModulePath = "${testModulePath}{0}${env:PSModulePath}" -f [System.IO.Path]::PathSeparator
+            $scriptBlock = [scriptblock]::Create("Invoke-Pester -Path $testScripts -OutputFormat NUnitXml -OutputFile $testResultsFile -Show Describe")
```

Comment:
I'd be OK with showing only the `Describe` output but then Pester does not show the summary any more, which we definitely need if we want to avoid scrolling through all test results when running locally. Can we therefore use the `Describe,Summary` option please